circuit_passes/src/passes/loop_unroll/body_extractor.rs

```diff
@@ -132,20 +154,37 @@ impl ExtraArgsResult {
     }
 }
 
-#[derive(Clone, Debug, Eq, PartialEq, Default)]
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd)]
+struct Key {
+    loop_bucket_id: BucketId,
+    iter_count: usize,
+    extra_arg_info: Rc<ExtraArgsResult>,
+}
+
+#[derive(Debug, Default)]
 pub struct LoopBodyExtractor {
-    new_body_functions: RefCell<Vec<FunctionCode>>,
+    new_body_functions: RefCell<BTreeMap<Key, FunctionCode>>,
+    /// Exists only to stabilize function order for lit test output.
+    func_creation_order: RefCell<Vec<String>>,
```

Comment:
If we care about the performance impact of this, we would need to add a cmd line flag for lit testing I think.

Reply:
We could go back to a manual CI script then, to get more control. I honestly don't like leaving nix handle everything